mewarpx/mewarpx/diags_store/diag_base.py

```diff
@@ -194,21 +196,26 @@ def __init__(self, diag_steps, preset_string='default',
         if install:
             callbacks.installafterstep(self.text_diag)
 
+        self.particle_steps_total = 0
+        self.previous_particle_steps_total = 0
+
     def text_diag(self):
         """Write requested information to output."""
+        live_parts, parts_per_species_str = self._get_part_nums()
+        self.particle_steps_total += live_parts
```

Comment:
I am still concerned about this. The `_get_part_nums()` function involves a parallel AllReduce (in `AMReX_ParticleContainerI.H` via `TotalNumberOfParticles`), doing this at every step seems like a somewhat big cost for very little benefit. So I would still default to the less resolved measurement described in https://github.com/ModernElectron/WarpX/pull/68#discussion_r708613807

Reply:
Because the results are mostly the same, I think this is a good change